enclave/cert.c

```diff
@@ -327,19 +321,51 @@ static mbedtls_x509_crt* _sort_certs_by_issue_date(mbedtls_x509_crt* chain)
     return sorted;
 }
 
+/* Call mbedlts_x509_crt_verify and handle error logging */
+static oe_result_t _mbedtls_x509_crt_verify(
+    mbedtls_x509_crt* leaf_cert,
+    mbedtls_x509_crt* ca_cert_chain,
+    mbedtls_x509_crl* ca_crls)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    uint32_t flags = 0;
+    if (mbedtls_x509_crt_verify(
+            leaf_cert, ca_cert_chain, ca_crls, NULL, &flags, NULL, NULL) != 0)
+    {
+        char error[1024] = {0};
+        mbedtls_x509_crt_verify_info(error, sizeof(error), "", flags);
+        if (flags & MBEDTLS_X509_BADCERT_REVOKED)
+            result = OE_VERIFY_REVOKED;
+        else if (flags & MBEDTLS_X509_BADCRL_EXPIRED)
+            result = OE_VERIFY_CRL_EXPIRED;
+        else
+            result = OE_VERIFY_FAILED;
```

Comment:
Maybe there are other MBEDTLS errors worth catching (like mismatch?), but probably good for now since you will at least see the error message in the log.

Reply:
There's a fairly long list of errors that it could return and I opted not to inject all of them into our error codes. Even the intersection set with OpenSSL is pretty big (10+). I don't want to reserve error codes for them unless we actually exercise the code paths that test for them, which we don't explicitly verify today.

Reply:
Yeah, I think we can expand as needed for clarification.